health/manager.go

```diff
@@ -0,0 +1,17 @@
+package health
+
+//go:generate mockery --name=Manager --filename=manager.go --output=../mocks/health
+
+type Manager interface {
```

Comment:
I also thought about calling this a checker, maybe not everything needs to necessarily be a manager?

I also thought about defining the interface in the api package, to remove the api packages dependency on this package, but we don't really do that elsewhere in our code. Might be something to look at in the future if we should define the interfaces where they're used to mitigate the number of dependencies in certain packages?